### PR TITLE
[MRG] Add x-ref to Projection object and to scipy, improve formatting with double backticks

### PR DESCRIPTION
**Pull request by mscheltienne**
A couple of doc improvements, after I looked at the `compute_proj_xxx` and `raw.filter` docstring.

### CODE REVIEW

File: mne/utils/docs.py [hunk 1]
```diff
@@ -1983,7 +1983,7 @@ def _reflow_param_docstring(docstring, has_first_line=True, width=75):
 ] = """
 iir_params : dict | None
     Dictionary of parameters to use for IIR filtering.
-    If iir_params is None and method="iir", 4th order Butterworth will be used.
+    If `i`ir_params=None`` and ``method="iir"``, 4th order Butterworth will be used.
```
Comment (mscheltienne) [inline mne/utils/docs.py:1986]:
```suggestion
    If ``iir_params=None`` and ``method="iir"``, 4th order Butterworth will be used.
```

**Review by hoechenberger — APPROVED** [review]
LGTM!

File: doc/conf.py [hunk 2]
```diff
@@ -184,8 +184,9 @@
 docscrape.ClassDoc.extra_public_methods = mne.utils._doc_special_members
 numpydoc_class_members_toctree = False
 numpydoc_show_inherited_class_members = {
-    "mne.SourceSpaces": False,
     "mne.Forward": False,
+    "mne.Projection": False,
```
Comment (mscheltienne) [inline doc/conf.py:188]: I don't think it makes sense to document the `dict` methods for a `Projection`. Users are not suppose to do `Projection.update(...)` anyway.

File: mne/_fiff/proj.py [hunk 3]
```diff
@@ -39,9 +39,10 @@
 
 
 class Projection(dict):
-    """Projection vector.
+    """Dictionary-like object holding a projection vector.
 
-    A basic class to proj a meaningful print for projection vectors.
+    Projection vectors are stored in a list in ``inst.info["projs"]``. Each projection
+    vector has 5 keys: ``active``, ``data``, ``desc``, ``explaied_var``, ``kind``.
```
Comment (mscheltienne) [inline mne/_fiff/proj.py:45]:
```suggestion
    vector has 5 keys: ``active``, ``data``, ``desc``, ``explained_var``, ``kind``.
```

File: mne/utils/docs.py [hunk 4]
```diff
@@ -3130,9 +3130,9 @@ def _reflow_param_docstring(docstring, has_first_line=True, width=75):
     suppression.
     When ``method='iir'``, ``phase='zero'`` (default) or
     ``phase='zero-double'`` constructs and applies IIR filter twice, once
-    forward, and once backward (making it non-causal) using filtfilt.
+    forward, and once backward (making it non-causal) using `~scipy.signal.filtfilt`.
```
Comment (larsoner) [inline mne/utils/docs.py:3133]: At some point we discussed going back to being explicit on these like
```suggestion
    forward, and once backward (making it non-causal) using :func:`~scipy.signal.filtfilt`.
```
so we could potentially change our `default_role` at a later date

Comment (drammock) [inline mne/utils/docs.py:3133]: without the `:func:` prefix, the parentheses aren't rendered.

```suggestion
    forward, and once backward (making it non-causal) using
    :func:`~scipy.signal.filtfilt`.
```

File: mne/utils/docs.py [hunk 5]
```diff
@@ -3130,9 +3130,9 @@ def _reflow_param_docstring(docstring, has_first_line=True, width=75):
...
     If ``phase='forward'``, it constructs and applies forward IIR filter using
-    lfilter.
+    `~scipy.signal.lfilter`.
```
Comment (drammock) [inline mne/utils/docs.py:3135]:
```suggestion
    :func:`~scipy.signal.lfilter`.
```

File: mne/utils/docs.py [hunk 6]
```diff
@@ -2511,8 +2511,8 @@ def _reflow_param_docstring(docstring, has_first_line=True, width=75):
     "method_fir"
 ] = """
 method : str
-    'fir' will use overlap-add FIR filtering, 'iir' will use IIR
-    forward-backward filtering (via filtfilt).
+    ``'fir'`` will use overlap-add FIR filtering, ``'iir'`` will use IIR
+    forward-backward filtering (via `~scipy.signal.filtfilt`).
```
Comment (drammock) [inline mne/utils/docs.py:2515]:
```suggestion
    forward-backward filtering (via :func:`~scipy.signal.filtfilt`).
```

**Review by drammock — APPROVED** [review]
just a couple nitpicks

File: mne/utils/docs.py (hunk 4 above)
Comment (mscheltienne) [reply mne/utils/docs.py:3133]: Sure, but just to know, what does it changes? It doesn't fail to resolve, with or without the explicit `:func:`, `:meth:`, `:class:`, ...

Comment (drammock) [reply mne/utils/docs.py]: see my comment.  It doesn't render parentheses without the explicit role

Comment (mscheltienne) [reply mne/utils/docs.py]: OK disregard that comment, I misread 

> so we could potentially change our default_role at a later date

And it makes sense to be working for now with `default_role = "py:obj"`.

Comment (mscheltienne) [reply mne/utils/docs.py]: Thanks! I'm reading too much in diagonals.. :see_no_evil:

**Pull request merged**